Ryujinx/Ui/ApplicationLibrary.cs

```diff
@@ -0,0 +1,374 @@
+                                controlFs = GetControlFs(xci.OpenPartition(XciPartitionType.Secure));
+                            }
+                            else { controlFs = GetControlFs(new PartitionFileSystem(file.AsStorage())); }
+
+                            // Creates NACP class from the NACP file
+                            IFile controlNacp = controlFs.OpenFile("/control.nacp", OpenMode.Read);
+                            Nacp  controlData = new Nacp(controlNacp.AsStream());
+
+                            // Get the title name, title ID, developer name and version number from the NACP
+                            version = controlData.DisplayVersion;
+
+                            titleName = controlData.Descriptions[(int)DesiredTitleLanguage].Title;
+                            if (string.IsNullOrWhiteSpace(titleName))
+                            {
+                                titleName = controlData.Descriptions.ToList().Find(x => !string.IsNullOrWhiteSpace(x.Title)).Title;
+                            }
+
+                            titleId = controlData.PresenceGroupId.ToString("x16");
+                            if (string.IsNullOrWhiteSpace(titleId)) { titleId = controlData.SaveDataOwnerId.ToString("x16"); }
+                            if (string.IsNullOrWhiteSpace(titleId)) { titleId = (controlData.AddOnContentBaseId - 0x1000).ToString("x16"); }
+
+                            developer = controlData.Descriptions[(int)DesiredTitleLanguage].Developer;
+                            if (string.IsNullOrWhiteSpace(developer))
+                            {
+                                developer = controlData.Descriptions.ToList().Find(x => !string.IsNullOrWhiteSpace(x.Developer)).Developer;
+                            }
+
+                            // Read the icon from the ControlFS and store it as a byte array
+                            try
+                            {
+                                IFile icon = controlFs.OpenFile($"/icon_{DesiredTitleLanguage}.dat", OpenMode.Read);
+                                using (MemoryStream ms = new MemoryStream())
+                                {
+                                    icon.AsStream().CopyTo(ms);
+                                    applicationIcon = ms.ToArray();
+                                }
+                            }
+                            catch (HorizonResultException)
+                            {
+                                IDirectory controlDir = controlFs.OpenDirectory("./", OpenDirectoryMode.All);
+                                foreach (DirectoryEntry entry in controlDir.Read())
+                                {
+                                    if (entry.Name == "control.nacp") { continue; }
+
+                                    IFile icon = controlFs.OpenFile(entry.FullPath, OpenMode.Read);
+                                    using (MemoryStream ms = new MemoryStream())
```

Comment:
```suggestion
                                    using (MemoryStream stream = new MemoryStream())
```
use an explicit name is maybe better.